src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -6815,10 +7054,18 @@ public Expression SetValue(Compiler compiler, Expression rhs)
             Expression setExpr;
             if (arrayLiteral != null)
             {
+                var expressions = new Expression[arrayLiteral.Elements.Count + 2];
+                expressions[0] = targetExpr;
+                expressions[^1] = temp;
+                for (int i = 1, j = 0; i < arrayLiteral.Elements.Count; i++, j++)
```

Comment:
This one is incorrect.
```suggestion
                for (int i = 1, j = 0; j < arrayLiteral.Elements.Count; i++, j++)
```